databricks/koalas/frame.py

```diff
@@ -4325,6 +4325,113 @@ def op(kser):
+        core.window.Expanding.cov : Expanding sample covariance.
+        core.window.Rolling.cov : Rolling sample covariance.
+        Notes
+        -----
+        Returns the covariance matrix of the DataFrame's time series.
+        The covariance is normalized by N-ddof.
+        For DataFrames that have Series that are missing data (assuming that
+        data is `missing at random
+        <https://en.wikipedia.org/wiki/Missing_data#Missing_at_random>`__)
+        the returned covariance matrix will be an unbiased estimate
+        of the variance and covariance between the member Series.
+        However, for many applications this estimate may not be acceptable
+        because the estimate covariance matrix is not guaranteed to be positive
+        semi-definite. This could lead to estimate correlations having
+        absolute values which are greater than one, and/or a non-invertible
+        covariance matrix. See `Estimation of covariance matrices
+        <https://en.wikipedia.org/w/index.php?title=Estimation_of_covariance_
+        matrices>`__ for more details.
+        Examples
+        --------
+        >>> kdf = ks.DataFrame([(1, 2), (0, 3), (2, 0), (1, 1)],
+        ...                   columns=['dogs', 'cats'])
+        >>> kdf.cov()
+                  dogs      cats
+        dogs  0.666667 -1.000000
+        cats -1.000000  1.666667
+        >>> np.random.seed(42)
+        >>> kdf = ks.DataFrame(np.random.randn(1000, 5),
+        ...                   columns=['a', 'b', 'c', 'd', 'e'])
+        >>> kdf.cov()
+                  a         b         c         d         e
+        a  0.998438 -0.020161  0.059277 -0.008943  0.014144
+        b -0.020161  1.059352 -0.008543 -0.024738  0.009826
+        c  0.059277 -0.008543  1.010670 -0.001486 -0.000271
+        d -0.008943 -0.024738 -0.001486  0.921297 -0.013692
+        e  0.014144  0.009826 -0.000271 -0.013692  0.977795
+
+        **Minimum number of periods**
+
+        This method also supports an optional ``min_periods`` keyword
+        that specifies the required minimum number of non-NA observations for
+        each column pair in order to have a valid result:
+        >>> np.random.seed(42)
+        >>> kdf = ks.DataFrame(np.random.randn(20, 3),
+        ...                   columns=['a', 'b', 'c'])
+        >>> kdf.loc[:5, 'a'] = np.nan
+        >>> kdf.loc[5:10, 'b'] = np.nan
+        >>> kdf.cov(min_periods=12)
+                  a         b         c
+        a  0.331350       NaN -0.148156
+        b       NaN  1.008785  0.164205
+        c -0.148156  0.164205  0.895202
+        """
+        num_cols = [
+            label
+            for label in self._internal.column_labels
+            if isinstance(self._internal.spark_type_for(label), (NumericType))
+        ]
+        kdf = self[num_cols]
+        names = [name for t in num_cols for name in t]
+        mat = kdf.to_pandas().to_numpy(dtype=float, copy=False)
```

Comment:
I'm afraid using `to_pandas()` without any restriction is not a good idea. It will cause OOM if the data side doesn't fit in a driver's memory.